### ISSUE TITLE
'host_specific_hostname_identity' is undefined

### ISSUE THREAD

**Issue by DrMoriarty** [spantaleev/matrix-docker-ansible-deploy#1]
Hello!

I configured ansible as described in Readme file. When I run `ansible-playbook -i inventory/hosts setup.yml --tags=setup-main` I always get the error:

```
TASK [matrix-server : Determine domains to obtain certificates for (Matrix)] ***********************************************************************************************
fatal: [matrix.mobilap.ru]: FAILED! => {"failed": true, "msg": "the field 'args' has an invalid value, which appears to include a variable that is undefined. The error was: matrix.{{ hostname_identity }}: {{ host_specific_hostname_identity }}: 'host_specific_hostname_identity' is undefined\n\nThe error appears to have been in '/Volumes/ST1000/projects/WVV2/matrix-docker-ansible-deploy/roles/matrix-server/tasks/setup_ssl.yml': line 3, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Determine domains to obtain certificates for (Matrix)\n  ^ here\n"}

PLAY RECAP *****************************************************************************************************************************************************************
matrix.mobilap.ru          : ok=11   changed=0    unreachable=0    failed=1  
```

I double checked that my vars.yml is correct and it contains `host_specific_hostname_identity ` variable

**Comment by DrMoriarty** [comment]
The only solution I have found is to add definitions of `host_specific_matrix_ssl_support_email` and `host_specific_hostname_identity` to `roles/matrix-server/defaults/main.yml`

**Comment by spantaleev** [comment]
Is your inventory host file (`inventory/matrix.<your-domain>/vars.yml`) based on the example file (`examples/host-vars.yml`) like the README says?

In that case, your inventory hostname file (`inventory/matrix.<your-domain>/vars.yml`) should already contain definitions for `host_specific_matrix_ssl_support_email` and `host_specific_hostname_identity`.

So unless you added these variables **once again** in `roles/matrix-server/defaults/main.yml`, it was showing that error?

Which version of Ansible is this and under which OS/distro?

**Comment by DrMoriarty** [comment]
Yes, I copied `examples/host-vars.yml` to `inventory/matrix.mobilap.ru/vars.yml` then I set the values of this two parameters.

The error was in every run until I set this two additional variables in `roles/matrix-server/defaults/main.yml`

```
bash-3.2$ ansible --version
ansible 2.3.2.0
  config file = /Volumes/ST1000/projects/WVV2/matrix-docker-ansible-deploy/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.14 (default, Sep 22 2017, 00:06:07) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
```

MacOSX 10.12.6 (16G29)

**Comment by spantaleev** [comment]
Seems like the README instructions were incorrect.

They instructed you to create an `inventory/matrix.<host>` directory, but it actually should have been `inventory/host_vars/matrix.<host>` (there needs to be an extra `host_vars` directory level there).

Because of this, the host variables were not found and read by Ansible.

I have fixed the incorrect README instructions in edd97d33c14be.

If you do a `git pull` and `mv inventory/matrix.mobilap.ru inventory/host_vars/matrix.mobilap.ru` you should be able to get it solved.

Thanks for reporting the problem, so we could get it fixed for everyone!
